sim-lib/src/sim_node.rs

```diff
@@ -229,23 +229,42 @@ impl SimulatedChannel {
         }
     }
 
-    /// Adds a htlc to the appropriate side of the simulated channel, checking its policy and balance are okay.
-    fn add_htlc(&mut self, node: PublicKey, htlc: Htlc) -> Result<(), ForwardingError> {
-        if htlc.amount_msat == 0 {
-            return Err(ForwardingError::ZeroAmountHtlc);
+    fn get_mut_node(&mut self, pubkey: &PublicKey) -> Result<&mut ChannelState, ForwardingError> {
```

Comment:
nit: the usual naming convention for mutable methods is to have `_mut` at the end

refs:

https://doc.rust-lang.org/std/collections/struct.HashMap.html#method.get_mut
https://doc.rust-lang.org/std/borrow/trait.BorrowMut
https://doc.rust-lang.org/std/ops/trait.DerefMut.html
https://doc.rust-lang.org/std/ops/trait.IndexMut.html